structuralcodes/codes/ec2_2004/shear.py

```diff
@@ -1,502 +1,539 @@
+        fyk (float): The characteristic strength of the reinforcement
+            steel in MPa.
+
+    Kwargs:
+        alpha (float): The angle of the shear reinforcement with
+            respect to the neutral axis in degrees. Default value = 90
+            degrees.
+        gamma_s (float): Partial factor of the reinforcement steel.
+            Default value = 1.15. Value might differ between
+            National Annexes.
+
+    Returns:
+        float: The shear resistance of the shear reinforcement in N.
+
+    Raises:
+        ValueError: When theta < 21.8 degrees or theta > 45 degrees.
+    """
+    fywd = fyk / gamma_s
+    _theta(theta)
+    theta = math.radians(theta)
+    alpha = math.radians(alpha)
+    return (
+        Asw
+        / s
+        * z
+        * fywd
+        * (1 / math.tan(theta) + 1.0 / math.tan(alpha))
+        * math.sin(alpha)
+    )
+
+
+# Equation (6.9 & 6.14)
+# For alpha == 90 degrees, Equation (6.14) reduces to Equation (6.9).
+def VRdmax(
+    bw: float,
+    z: float,
+    fck: float,
+    theta: float,
+    NEd: float,
+    Ac: float,
+    gamma_c: float = 1.5,
+    alpha: float = 90.0,
+    alpha_cc: float = 1.0,
+    limit_fyd: bool = False,
+) -> float:
+    """Calculate the maximum shear strength of the compression strut.
+
+    EN 1992-1-1 (2005). Eq. (6.9)
+
+    Args:
+        bw (float): The smallest width of the cross-section in tension
+            in mm.
+        z (float): The inner lever arm of internal forces in mm.
+        fck (float): The characteristic compressive strength in MPa.
+        theta (float): The angle of the compression strut in degrees.
+        NEd (float): The normal force in the cross-section due to
+            loading or prestress (NEd > 0 for compression) in N.
+        Ac (float): The cross-sectional area of the concrete in mm2.
+
+    Kwargs:
+        gamma_c (float): Partial factor for concrete. Default value =
+            1.5, value might differ between National Annexes.
+        alpha (float): The angle of the shear reinforcement with
+            respect to the neutral axis in degrees. Default value = 90
+            degrees.
+        alpha_cc (float): A factor for considering long-term effects on the
+            strength, and effects that arise from the way the load is applied.
+        limit_fyd (bool): Flag to indicate if the design yield stress is
+            limited to 0.8 * fyk or not. This controls whether the stress
+            reduction factor of concrete is given by Eq. (6.6) (False) or
+            (6.10) (True).
+
+    Returns:
+        float: The shear strength of the shear reinforcement in N.
+
+    Raises:
+        ValueError: When theta < 21.8 degrees or theta > 45 degrees.
+        ValueError: When sigma_cp > fcd.
+    """
+    fcd = fcdfun(fck, alpha_cc=alpha_cc, gamma_c=gamma_c)
+    _theta(theta)
+    theta = math.radians(theta)
+    alpha = math.radians(alpha)
+    strength_reduction = v(fck) if not limit_fyd else v1(fck)
```

Comment:
Here we calculate the correct strength reduction factor according to the `limit_fyd` flag.